### PR TITLE
Update npm packages including express

### PR DESCRIPTION
**Pull request by irfanh94**
## Description
This PR updates several npm packages, with the main focus on updating the `express` package to its latest version. It ensures compatibility with the updated dependencies and applies minor fixes where necessary to avoid potential issues caused by outdated packages.

## Related Issue
No specific issue is linked to this PR. This update is primarily to maintain the project's health by keeping dependencies up to date.

## Motivation and Context
Keeping dependencies up to date is important for security, performance, and compatibility reasons. The main motivation behind this PR is to upgrade `express` and ensure that the project continues to function smoothly with the latest version.

## How Has This Been Tested?
- I tested this update by running the test suite using `npm run test`, and all tests passed successfully.
- No additional changes to the functionality were made, so the existing tests cover the changes introduced.

## Types of changes
- [x] Dependency update (non-breaking change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by notaphplover** [comment]
Hey @irfanh94, thank you so much for your contribution. I just updated dependencies myself, but I did not update `eslint`. Would you mind updating this PR with the `eslint` updates? Thanks in advance

**Comment by rsaz** [comment]
- deps updated
- eslint configured and updated

cc: @notaphplover

File: package.json [hunk 1]
```diff
@@ -56,10 +20,48 @@
...
+  "homepage": "https://github.com/inversify/inversify-express-utils#readme",
+  "jsnext:main": "es/index.js",
+  "keywords": [
+    "InversifyJS",
+    "express"
+  ],
+  "dependencies": {
+    "express": "4.21.1",
+    "http-status-codes": "2.3.0",
+    "inversify": "^6.0.3"
```
Comment (notaphplover) [inline package.json:41]: This should be a peer dependency as well.

File: src/index.ts [hunk 2]
```diff
@@ -11,4 +11,4 @@ export * from './debug';
 export * from './httpResponseMessage';
 export * from './content/stringContent';
 export * from './content/jsonContent';
-export * from './content/httpContent';
+export * from './content/httpContent';
```
Comment (notaphplover) [inline src/index.ts:14]: Let's keep the line feed

File: package.json [hunk 3]
```diff
@@ -56,10 +20,48 @@
...
+  },
+  "devDependencies": {
+    "@types/async": "3.2.24",
+    "@types/cookie-parser": "1.4.7",
+    "@types/express": "4.17.21",
```
Comment (notaphplover) [inline package.json:46]: Let's allow any minor:

```
  "@types/express": "^4.17.21",
```

**Pull request merged**